src/index.mako.html

```diff
@@ -709,82 +709,66 @@
-        printUrl = adminUrlRegexp.test(printUrl) ? printUrl : location.protocol + '${print_url}';
-
-        var apiParam = getParam('api_url');
-        var apiEnv = getParam('api_env');
-        var apiUrlBase = '${api_url}';
-
-        if (apiEnv){
-          if (apiEnv == 'prod'){
-            host = '//api3.geo.admin.ch';
-          } else {
-            host = '//mf-chsdi3.' + apiEnv + '.bgdi.ch';
+        var s3pathname = '${s3basepath}' == '' ? pathname : '${s3basepath}';
+        var prtl = location.protocol;
+        // Because s3 is hosted on infra
+        var supportedEnv = ['dev', 'int', 'prod', 'ci'];
+        var staging = '${staging}';
+
+        // Determines if one is using custom prodUrl (for user env)
+        var userEnvRegexp = new RegExp('\/\/.*(\/[a-z]*)');
+        // Force env via parameter for testing purposes (before Varnish)
+        var env = getParam('env');
+        function setBackend(prodUrl, techUrl, overwriteParam) {
+          var url = prodUrl;
+          if (env && env != 'prod' && supportedEnv.indexOf(env) !== -1) {
+            url = techUrl + env + '${tech_suffix}';
+          } else if (staging != 'prod' && !userEnvRegexp.test(prodUrl)) {
+            url = techUrl + staging + '${tech_suffix}';
           }
-        } else {
-          host = '${api_url}';
+          var overwrite = prtl + getParam(overwriteParam);
+          url = adminUrlRegexp.test(overwrite) ? overwrite : url;
+          return url;
         }
 
-        if (host.slice(-1) == '/'){
-          host = host.slice(0, -1);
-        }
-
-        if (apiParam){
-          apiUrl = host + '/' + apiParam;
-        } else {
-          apiUrl = host;
-        }
-
-        var shopUrl = getParam('shop_url');
-        var shopUrlBase = '${shop_url}';
-
-        if (shopUrl){
-          shopUrl = shopUrl;
-          if (shopUrl == 'dev' || shopUrl == 'int'){
-            shopUrl = '//shop-bgdi.'+ shopUrl + '.bgdi.ch';
-          } else if (shopUrl == 'prod'){
-              shopUrl = '//beta.shop.swisstopo.ch';
-          } else if (shopUrl == 'puzzle'){
-              shopUrl = '//shop-shopbgdi.openshift.puzzle.ch';
-          }
-        } else {
-          shopUrl = shopUrlBase;
-        }
+        var defaultApiUrl = prtl + '${api_url}';
+        var apiUrl = setBackend(defaultApiUrl, prtl + '${api_tech_url}', 'api_url');
+        var mapproxyUrl = setBackend(prtl + '${mapproxy_url}', prtl + '${mapproxy_tech_url}', 'mapproxy_url');
+        var wmsUrl = setBackend(prtl + '${wms_url}',  prtl + '${wms_tech_url}', 'wms_url');
+        var shopUrl = setBackend(prtl + '${shop_url}', prtl + '${shop_tech_url}', 'shop_url');
+        var publicUrl = setBackend(prtl + '${public_url}', prtl + '${public_tech_url}', 'public_url');
+        var printUrl = setBackend(prtl + '${print_url}', prtl + '${print_tech_url}', 'print_url');
+        var apiOverwrite = !!(getParam('api_url') || env || staging == 'dev');
```

Comment:
It is, at least here: 
https://mf-geoadmin3.dev.bgdi.ch/ltjeg/?topic=ech&lang=en&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege&layers_visibility=false,false,false,false&layers_timestamp=18641231,,,

And we build the cache on each build.

It's not the case on dev main now, because in dev main we probably have 
this branch...

I think we should stick to it

On 18.08.2016 15:14, Gasser Loïc wrote:
> In src/index.mako.html [1]:
> 
>> - } else if (shopUrl == 'prod'){
>> - shopUrl = '//beta.shop.swisstopo.ch';
>> - } else if (shopUrl == 'puzzle'){
>> - shopUrl = '//shop-shopbgdi.openshift.puzzle.ch';
>> - }
>> - } else {
>> - shopUrl = shopUrlBase;
>> - }
>> + var defaultApiUrl = prtl + '${api_url}';
>> + var apiUrl = setBackend(defaultApiUrl, prtl + '${api_tech_url}',
> 'api_url');
>> + var mapproxyUrl = setBackend(prtl + '${mapproxy_url}', prtl +
> '${mapproxy_tech_url}', 'mapproxy_url');
>> + var wmsUrl = setBackend(prtl + '${wms_url}', prtl +
> '${wms_tech_url}', 'wms_url');
>> + var shopUrl = setBackend(prtl + '${shop_url}', prtl +
> '${shop_tech_url}', 'shop_url');
>> + var publicUrl = setBackend(prtl + '${public_url}', prtl +
> '${public_tech_url}', 'public_url');
>> + var printUrl = setBackend(prtl + '${print_url}', prtl +
> '${print_tech_url}', 'print_url');
>> + var apiOverwrite = !!(getParam('api_url') || env || staging ==
> 'dev');
> 
> not the case at the moment.
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [2], or mute the
> thread [3].
> 
> 	*
> 
> Links:
> ------
> [1] 
> https://github.com/geoadmin/mf-geoadmin3/pull/3410#discussion_r75304936
> [2]
> https://github.com/geoadmin/mf-geoadmin3/pull/3410/files/67f005c7445b5c963a04ed238e8d53e9f6b62b8d#r75304936
> [3]
> https://github.com/notifications/unsubscribe-auth/ACdAHdkeN3sx99Ivo-f8qKjOr8r9qvYFks5qhFq7gaJpZM4JS-XH